File: csharp/Creational/01-FactoryMethod/Concrete Products/ElectricalJobCard .cs
public class ElectricalJobCard : IJobCard
{
    public string JobId    { get; } = Guid.NewGuid().ToString()[..8];
    public string Title    { get; }
    public string Category => "Electrical";
    public string Priority { get; }

    public ElectricalJobCard(string title, string priority)
    {
        Title    = title;
        Priority = priority;
    }

    public void PrintSummary() =>
        Console.WriteLine(
            $"[{Category}] #{JobId} — {Title} (Priority: {Priority})" +
            $"\n  → Requires: Certified electrician + safety inspection");
}

File: typescript/Creational/01-FactoryMethod/Concrete Products/ElectricalJobCard.ts
class ElectricalJobCard implements JobCard {
  readonly jobId    = shortId();
  readonly category = "Electrical";

  constructor(
    readonly title:    string,
    readonly priority: string
  ) {}

  printSummary(): void {
    console.log(`[${this.category}] #${this.jobId} — ${this.title} (Priority: ${this.priority})`);
    console.log("  → Requires: Certified electrician + safety inspection");
  }
}